Skip Wasm tests in run-javascriptcore-tests and run-jsc-stress-tests when Wasm is not enabled.

MenloDorian opened the pull request:
#### 25747ea0ed52aaf6665d5532df4d55fbe796517a
<pre>
Skip Wasm tests in run-javascriptcore-tests and run-jsc-stress-tests when Wasm is not enabled.
<a href="https://bugs.webkit.org/show_bug.cgi?id=298164">https://bugs.webkit.org/show_bug.cgi?id=298164</a>
<a href="https://rdar.apple.com/159541255">rdar://159541255</a>

Reviewed by Sosuke Suzuki.

This patch changes the following:
1. In run-javascriptcore-tests, skip wasm.yaml wholesale if Wasm is not supported.
2. In run-jsc-stress-tests, skip any tests that have &quot;wasm-&quot; or &quot;-wasm&quot; in the test name if Wasm
   is not supported.
3. Move some tests that are only relevant for Wasm from JSTests/stress to JSTests/wasm/stress/.

With these changes, CLoop builds no longer run nor fail wasm tests without us needing to manually
add skip statements or explicit checks for the WebAssembly object in the tests.

* JSTests/microbenchmarks/exceptions-simd.js:
* JSTests/microbenchmarks/interpreter-wasm.js:
* JSTests/microbenchmarks/memcpy-wasm-large.js:
* JSTests/microbenchmarks/memcpy-wasm-medium.js:
* JSTests/microbenchmarks/memcpy-wasm-small.js:
* JSTests/microbenchmarks/memcpy-wasm.js:
* JSTests/microbenchmarks/wasm-cc-int-to-int.js:
* JSTests/stress/hoist-get-wasm-exports.js:
* JSTests/stress/shared-wasm-memory-with-zero-byte.js:
* JSTests/wasm/noJIT/noJIT.js:
* JSTests/wasm/stress/cc-int-to-int-cross-module-with-exception.js:
* JSTests/wasm/stress/enumerator-oom.js: Renamed from JSTests/stress/enumerator-oom.js.
(func_1_):
* JSTests/wasm/stress/inline-simd-function.js:
* JSTests/wasm/stress/only-referenced.js:
* JSTests/wasm/stress/simple-inline-stacktrace-2.js:
* JSTests/wasm/stress/simple-inline-stacktrace-with-catch-2.js:
* JSTests/wasm/stress/strcat-bigint-oom.js: Renamed from JSTests/stress/strcat-bigint-oom.js.
(foo):
* JSTests/wasm/stress/throw-from-wasm-catch-in-baseline-JIT.js:
* JSTests/wasm/stress/webassembly-jstag-cache.js: Renamed from JSTests/stress/webassembly-jstag-cache.js.
(shouldBe):
(throw.new.Error):
* Tools/Scripts/run-javascriptcore-tests:
(runJSCStressTests):
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: <a href="https://commits.webkit.org/299431@main">https://commits.webkit.org/299431@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0bce0cd334831b4c2f2925fb6a048bd4484975f7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118850 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38531 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29182 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125039 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70912 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9723ad72-9daf-42e4-a368-5da5768d1951) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120728 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39227 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47113 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90200 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59715 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4ee49181-392a-48d7-a632-cd5ee7791304) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121803 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106556 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70707 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a8204547-c248-47a2-8085-aad4b14e9c47) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30325 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24670 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68698 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110968 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100706 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24857 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128090 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/117364 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45757 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34556 "Found 1 new test failure: imported/w3c/web-platform-tests/fetch/api/crashtests/body-window-destroy.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98857 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46123 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102776 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98637 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44083 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22087 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42317 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18943 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45627 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/146062 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45092 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37570 "Found 1 new JSC binary failure: testapi, Found 18622 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default, ChakraCore.yaml/ChakraCore/test/Function/StackArgsWithFormals.js.default, ChakraCore.yaml/ChakraCore/test/Function/apply.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48437 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46777 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

sosukesuzuki commented on Tools/Scripts/run-jsc-stress-tests:
```diff
@@ -2567,7 +2582,8 @@ def handleCollectionDirectory(collection)
             | path |
-            
+
+            next if not $isWasmPlatformx and isWasmTest(path)
```
typo: `$isWasmPlatformx` should be `$isWasmPlatform`

MenloDorian replied:
Thanks.  Will fix.

sosukesuzuki commented on Tools/Scripts/run-jsc-stress-tests:
```diff
@@ -2538,7 +2552,8 @@ def handleCollectionFile(collection)
                     | path |
-                    
+
+                    next if not $isWasmPlatform and isWasmPath(path)
```
`isWasmPath` is not defined so we should use `isWasmTest` ?

MenloDorian replied:
Thanks.

sosukesuzuki reviewed (approved):
r=me

Commit:
Skip Wasm tests in run-javascriptcore-tests and run-jsc-stress-tests when Wasm is not enabled.
https://bugs.webkit.org/show_bug.cgi?id=298164
rdar://159541255

Reviewed by Sosuke Suzuki.

This patch changes the following:
1. In run-javascriptcore-tests, skip wasm.yaml wholesale if Wasm is not supported.
2. In run-jsc-stress-tests, skip any tests that have "wasm-" or "-wasm" in the test name if Wasm
   is not supported.
3. Move some tests that are only relevant for Wasm from JSTests/stress to JSTests/wasm/stress/.

With these changes, CLoop builds no longer run nor fail wasm tests without us needing to manually
add skip statements or explicit checks for the WebAssembly object in the tests.

* JSTests/microbenchmarks/exceptions-simd.js:
* JSTests/microbenchmarks/interpreter-wasm.js:
* JSTests/microbenchmarks/memcpy-wasm-large.js:
* JSTests/microbenchmarks/memcpy-wasm-medium.js:
* JSTests/microbenchmarks/memcpy-wasm-small.js:
* JSTests/microbenchmarks/memcpy-wasm.js:
* JSTests/microbenchmarks/wasm-cc-int-to-int.js:
* JSTests/stress/hoist-get-wasm-exports.js:
* JSTests/stress/shared-wasm-memory-with-zero-byte.js:
* JSTests/wasm/noJIT/noJIT.js:
* JSTests/wasm/stress/cc-int-to-int-cross-module-with-exception.js:
* JSTests/wasm/stress/enumerator-oom.js: Renamed from JSTests/stress/enumerator-oom.js.
(func_1_):
* JSTests/wasm/stress/inline-simd-function.js:
* JSTests/wasm/stress/only-referenced.js:
* JSTests/wasm/stress/simple-inline-stacktrace-2.js:
* JSTests/wasm/stress/simple-inline-stacktrace-with-catch-2.js:
* JSTests/wasm/stress/strcat-bigint-oom.js: Renamed from JSTests/stress/strcat-bigint-oom.js.
(foo):
* JSTests/wasm/stress/throw-from-wasm-catch-in-baseline-JIT.js:
* JSTests/wasm/stress/webassembly-jstag-cache.js: Renamed from JSTests/stress/webassembly-jstag-cache.js.
(shouldBe):
(throw.new.Error):
* Tools/Scripts/run-javascriptcore-tests:
(runJSCStressTests):
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: https://commits.webkit.org/299431@main
